lib/modules/ens/embarkjs.js

```diff
@@ -171,58 +171,60 @@ __embarkENS.setProvider = function (config) {
   });
 };
 
-__embarkENS.resolve = function (name, callback) {
-  callback = callback || function () {};
-  if (!this.ens) {
-    return callback(providerNotSetError);
-  }
-  let node = namehash.hash(name);
-
-  function cb(err, addr) {
-    if (err === NoDecodeAddrError) {
-      return callback(name + " is not registered", "0x");
-    }
-    callback(err, addr);
-  }
-
-  return this.ens.methods.resolver(node).call((err, resolverAddress) => {
-    if (err) {
-      return cb(err);
+__embarkENS.resolve = function(name) {
```

Comment:
I am not sure we can remove the callback as it is part of the public API.
However we can support both for sure.

@iurimatias what do you think?

Reply:
This was my concern as well. However, on the other hand I wasn't sure whether `__embarkENS` really is a public facing API (for consumers). But it will be a breaking change indeed.

Reply:
In the last release (3.1.9) and documentation (https://embark.status.im/docs/naming_javascript.html) these functions do not take a callback and return a promise. The callback is a regression within `develop` branch. So removing the callback fixes a breaking change.

Reply:
@StevenJNPearce Good point right there! I just verified that 3.1.8 does not have `callback`, however `3.1.10` does. I guess this needs updates in the docs then (in case we want to keep this)

Looping in @iurimatias for another opinion on this.

Reply:
We should keep the callback. EmbarkJS expects its Naming plugins to have this param https://github.com/embark-framework/EmbarkJS/blob/master/src/names.js#L24 typically it's used by folks who are you using async.js instead of promises for e.g

Reply:
@iurimatias We should make this clear in the docs though, as pointed out in https://github.com/embark-framework/embark/pull/883/files#r219557791 this is not very obvious.

Reply:
I've created a PR for this here: https://github.com/embark-framework/embark-site/pull/76

Hope this makes it a bit more clear!